api/controllers/search.js

```diff
@@ -328,24 +353,60 @@ var searchCollection = async function (roles, projectPermissions, keywords, coll
     }
   });
 
-  aggregation.push({
-    $facet: {
-      searchResults: searchResultAggregation,
-      meta: [
-        {
-          $count: "searchResultsTotal"
-        }
-      ]
-    }
-  })
+	if (sortField !== 'projectTypes') {
+		aggregation.push({
+			$facet: {
+				searchResults: searchResultAggregation,
+				meta: [
+					{
+						$count: "searchResultsTotal"
+					}
+				]
+			}
+		})
+	}
 
   return new Promise(function (resolve, reject) {
     var collectionObj = mongoose.model(collection);
     collectionObj.aggregate(aggregation)
       .collation(collation)
       .exec()
       .then(function (data) {
-        resolve(data);
+				let collectionData;
+
+				// If the sort field is projectTypes, handle the sorting manually.
+        if ('projectTypes' === sortField) {
+					const rawResults = data || [];
+          rawResults.forEach(rr => {
+            const list = Array.isArray(rr.projectTypesFiltered) ? rr.projectTypesFiltered : [];
+            const sorted = list.slice().sort(); // alphabetically
+            rr._sortKey = sorted.join(', ');
+          });
+
+          const sortedResults = rawResults.sort((a, b) => {
+						if (a._sortKey < b._sortKey) return sortDirection === 1 ? -1 : 1;
+						if (a._sortKey > b._sortKey) return sortDirection === 1 ? 1 : -1;
```

Comment:
I feel like you could do some quick math to make this more concise. `sortDirection` will be either 1 or -1 right?

```suggestion
						if (a._sortKey > b._sortKey) return sortDirection * 1;
```

Reply:
Yes I used multiplication in my solution, I think you will find it much less verbose.